src/TLuaInterpreter.cpp

```diff
@@ -12876,7 +12876,7 @@ void TLuaInterpreter::ttsBuild()
     bSpeechQueueing = false;
 
     connect(speechUnit, &QTextToSpeech::stateChanged, &TLuaInterpreter::ttsStateChanged);
-
+    connect(speechUnit, &QTextToSpeech::localeChanged, &TLuaInterpreter::ttsLocaleChanged);
```

Comment:
Not sure if this is the signal you want to connect to - you might instead want to arrange to connect something to the `(void) mudlet::signal_guiLanguageChanged(const QString& language code)` SIGNAL and in that read the newly set value provided by the `(const QLocale&) mudlet::getUserLocale()` method. That way you will have the voices associated with the UI language.

Obviously we may need to check that the `QLocale` provided by `getUserLocale()` is the one that is used when the TTS system is initialised in this class anyhow...